resources/providers/cluster_provider.go

```diff
@@ -0,0 +1,58 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package providers
+
+import (
+	"context"
+	"fmt"
+	"github.com/elastic/cloudbeat/config"
+	"github.com/elastic/cloudbeat/resources/providers/awslib"
+	k8s "k8s.io/client-go/kubernetes"
+)
+
+type ClusterNameProviderAPI interface {
+	GetClusterName(ctx context.Context, cfg config.Config) (string, error)
+}
+
+type ClusterNameProvider struct {
+	KubernetesClusterNameProvider KubernetesClusterNameProviderApi
+	EKSMetadataProvider           awslib.MetadataProvider
+	EKSClusterNameProvider        awslib.ClusterNameProvider
+	KubeClient                    k8s.Interface
+	AwsConfigProvider             awslib.ConfigProviderAPI
+}
+
+func (provider ClusterNameProvider) GetClusterName(ctx context.Context, cfg config.Config) (string, error) {
+	switch cfg.Type {
+	case config.InputTypeVanillaK8s:
+		return provider.KubernetesClusterNameProvider.GetClusterName(cfg, provider.KubeClient)
+	case config.InputTypeEks:
+		awsConfig, err := provider.AwsConfigProvider.InitializeAWSConfig(ctx, cfg.AWSConfig)
+		if err != nil {
+			return "", fmt.Errorf("failed to initialize aws configuration for identifying the cluster name: %v", err)
+		}
+		metadata, err := provider.EKSMetadataProvider.GetMetadata(ctx, awsConfig)
+		if err != nil {
+			return "", fmt.Errorf("failed to get the ec2 metadata required for identifying the cluster name: %v", err)
+		}
+		instanceId := metadata.InstanceID
+		return provider.EKSClusterNameProvider.GetClusterName(ctx, awsConfig, instanceId)
+	default:
+		panic(fmt.Sprintf("unknown cluster type: %s", cfg.Type))
```

Comment:
The reason I put the `panic` here is that I want to fast fail when someone tries to use the cluster provider but didn't implement the relevant cluster provider.
I think that logs can be ignored and probably will be

WDYT?

Reply:
You've noted prior regarding the cluster name that:
> It is not crucial information for the findings 
Obviously, this particular code is problematic but I'm wondering if should even `panic` instead of ignoring and logging it as an error.

Reply:
@oren-zohar we wish to hear a 3rd opinion.

I am Ok with both of the approaches 👍

Reply:
Panic is typically used when your code could end up in a bad state that would cause it to crash; something that is truly unexpected. In cases there's no cluster name we're still supposed to give uuid to the cluster as we do right now, no? why would cloudbeat crash in that case? what if the user doesn't have a way to fix it? will it mean he will never be able to run cloudbeat?

Reply:
I'll try to explain.

The only case where a user will encounter a `panic`, is when the provided Kubernetes type is not supported.
This should never happen since the Kubernetes type only comes from the integration.
I thought about putting it here to "fail fast" - so that when a developer will use this new provider the program will immediately crush, then it will force the developer to implement the relevant cluster name provider/not use it.

When using a log, a developer will probably miss this.

That being said, if you prefer a log - I can do it

Reply:
Oh I see, makes sense, we shouldn't run without relevant inputType. I would make the error message more informative in that case, also I would refer to the input type and not the cluster type in the panic message so it's more clear for us to debug.